consensus/state.go

```diff
@@ -788,7 +814,10 @@ func (cs *State) receiveRoutine(maxSteps int) {
 			}
 
 			// handles proposals, block parts, votes
-			cs.handleMsg(mi)
+			if err := cs.handleMsg(mi); err != nil {
+				// TODO: why are we erroring on our own messages?
+				// should we panic ?
+			}
 
```

Comment:
Bro you can't make any assumptions about any properties of the `mi` you get from this chan except what's guaranteed by the language. handleMessage returns an error, you as a caller can't make any assumptions about _why_ that error occurred, all that matters is if there's an err or not. If there's an err then your operation failed. And there is basically no situation where panicking is an acceptable response to functions that return errors. 

If handleMsg fails, what does that mean, in the context? Should it terminate the goroutine that's running the receiveRoutine method of this type? If so, `return fmt.Errorf("handle message: %w", err)`. Should the receive goroutine be resilient to this error? If so, log the error and continnue.